src/stan/services/optimize/bfgs.hpp

```diff
@@ -0,0 +1,202 @@
+        model.constrained_param_names(names, true, true);
+        parameter_writer(names);
+
+        if (save_iterations) {
+          std::vector<double> values;
+          msg.str("");
+          model.write_array(rng, cont_vector, disc_vector, values,
+                            true, true, &msg);
+          if (msg.str().length() > 0)
+            message_writer(msg.str());
+
+          values.insert(values.begin(), lp);
+          parameter_writer(values);
+        }
+        int ret = 0;
+
+        while (ret == 0) {
+          interrupt();
+          if (refresh > 0
+              && (bfgs.iter_num() == 0
+                  || (bfgs.iter_num() + 1 % refresh == 0)))
+            message_writer("    Iter "
+                           "     log prob "
+                           "       ||dx|| "
+                           "     ||grad|| "
+                           "      alpha "
+                           "     alpha0 "
+                           " # evals "
+                           " Notes ");
+
+          ret = bfgs.step();
+          lp = bfgs.logp();
+          bfgs.params_r(cont_vector);
+
+          if (refresh > 0
+              && (ret != 0
+                  || !bfgs.note().empty()
+                  || bfgs.iter_num() == 0
+                  || (bfgs.iter_num() + 1 % refresh == 0))) {
+            msg.str("");
+            msg << " " << std::setw(7) << bfgs.iter_num() << " ";
+            msg << " " << std::setw(12) << std::setprecision(6)
+                << lp << " ";
+            msg << " " << std::setw(12) << std::setprecision(6)
+                << bfgs.prev_step_size() << " ";
+            msg << " " << std::setw(12) << std::setprecision(6)
+                << bfgs.curr_g().norm() << " ";
+            msg << " " << std::setw(10) << std::setprecision(4)
+                << bfgs.alpha() << " ";
+            msg << " " << std::setw(10) << std::setprecision(4)
+                << bfgs.alpha0() << " ";
+            msg << " " << std::setw(7)
+                << bfgs.grad_evals() << " ";
+            msg << " " << bfgs.note() << " ";
+            message_writer(msg.str());
+            msg.str("");
+          }
+
+          if (bfgs_ss.str().length() > 0) {
+            message_writer(bfgs_ss.str());
+            bfgs_ss.str("");
+          }
+
+          if (save_iterations) {
+            std::vector<double> values;
+            msg.str("");
+            model.write_array(rng, cont_vector, disc_vector, values,
+                              true, true, &msg);
+            if (msg.str().length() > 0)
+              message_writer(msg.str());
+
+            values.insert(values.begin(), lp);
+            parameter_writer(values);
+          }
+        }
+
+        if (!save_iterations) {
+          std::vector<double> values;
+          msg.str("");
+          model.write_array(rng, cont_vector, disc_vector, values,
+                            true, true, &msg);
+          if (msg.str().length() > 0)
```

Comment:
Will there be an easy way to find all these to fix?  I won't comment on any more, but the pattern is ubiquitous.